Dockerfile

```diff
@@ -1,29 +1,34 @@
-# Build container
+# Stage 1: Build container
+# Use a specific Golang version and Alpine Linux as the base image
 ARG GOVERSION=1.20.12
 ARG ALPINEVERSION=3.19
 
 FROM --platform=${BUILDPLATFORM} \
     golang:$GOVERSION-alpine${ALPINEVERSION} AS build
 
+# Set the working directory for the build
 WORKDIR /src
-RUN apk --no-cache add git build-base
 
-ENV GO111MODULE=on \
-    CGO_ENABLED=0
+# Install necessary dependencies for building
+RUN apk --no-cache -U add git build-base curl jq
 
-ARG VERSION=2023.10.0
-RUN git clone https://github.com/cloudflare/cloudflared --depth=1 --branch ${VERSION} .
+# Fetch the latest release version using GitHub API
+RUN VERSION=$(curl -s https://api.github.com/repos/cloudflare/cloudflared/releases/latest | jq -r .tag_name) \
+    && git clone https://github.com/cloudflare/cloudflared --depth=1 --branch ${VERSION} .
```

Comment:
This defeats the purpose of the docker tags, introduces future errors with no code changes and prevents version control of cloudflared through git history.
In other words, it conflicts with the goals of the project.

There are several examples of cloudflared updates which require changes to the Docker image or waiting for a new version in order for a subsequent change to be made before the image works.